daisy/workflow/validate.go

```diff
@@ -141,6 +142,9 @@ func (w *Workflow) validateRequiredFields() error {
 	if w.Name == "" {
 		return errors.New("must provide workflow field 'Name'")
 	}
+	if !rfc1035Rgx.MatchString(strings.ToLower(w.Name)) {
```

Comment:
What do you think about having the workflow not care about this? (I'm thinking in terms of decoupling workflow logic from resource stuff) We could have resource name generation sanitize the workflow name. That way, name enforcement is left to resource creation steps' implementation.